mergekit/plan.py

```diff
@@ -66,7 +67,10 @@ def __init__(
         self.arch_info = arch_info
         self.options = options
         self.out_model_config = out_model_config
-        self._method = merge_methods.get(config.merge_method)
+        if getattr(config, "merge_method", None):
+            self._method = merge_methods.get(config.merge_method)
+        elif getattr(config, "metric_method", None):
```

Comment:
Would be good to add a validator to MergeConfig that checks that exactly one of these fields is set.